### PR TITLE
fix incorrect encrypted backup restore error handling. Issue #10179

### PR DESCRIPTION
**Pull request by vktg**
- [ ] Redmine Issue: https://redmine.pfsense.org/issues/10179
- [ ] Ready for review

If you trying to restore encrypted backup, but select invalid file, you got blank page after pressing "restore" ( caused by 'return 1;' in code )
If you enter incorrect decryption password, you got error message: 
'A full configuration restore was selected but a pfsense tag could not be located.'
which is incorrect, it should be:
'File decryption failed. Incorrect password or file is invalid.'

### CODE REVIEW

File: src/usr/local/www/diag_backup.php [hunk 1]
```diff
@@ -233,18 +233,17 @@ function check_and_returnif_section_exists($section) {
 					/* read the file contents */
 					$data = file_get_contents($_FILES['conffile']['tmp_name']);
 					if (!$data) {
-						log_error(sprintf(gettext("Warning, could not read file %s"), $_FILES['conffile']['tmp_name']));
-						return 1;
-					}
-
-					if ($_POST['decrypt']) {
+						$input_errors[] = gettext("Warning, could not read file {$_FILES['conffile']['tmp_name']}");
+					} else if ($_POST['decrypt']) {
```
Comment (jim-p) [inline src/usr/local/www/diag_backup.php]: This should be `elseif` ([ref](https://docs.netgate.com/pfsense/en/latest/development/developer-style-guide.html#general-rules))

File: src/usr/local/www/diag_backup.php [hunk 2]
```diff
@@ -277,7 +276,7 @@ function check_and_returnif_section_exists($section) {
 								$savemsg = gettext("The configuration area has been restored. The firewall may need to be rebooted.");
 							}
 						}
-					} else {
+					} else if (!$input_errors) {
```
Comment (jim-p) [inline src/usr/local/www/diag_backup.php]: This should be `elseif` ([ref](https://docs.netgate.com/pfsense/en/latest/development/developer-style-guide.html#general-rules))

File: src/usr/local/www/diag_backup.php (hunk 1 above)
Comment (vktg) [reply src/usr/local/www/diag_backup.php]: fixed

File: src/usr/local/www/diag_backup.php (hunk 2 above)
Comment (vktg) [reply src/usr/local/www/diag_backup.php]: fixed

**Pull request merged**